README.md

```diff
@@ -7,3 +7,68 @@ If you just want to use the coupling between MOSAIC and ns-3, you can use the `n
+
+### Messaging protocol
+- Protobuf schema: ClientServerChannelMessages.proto
+- Framing: varint32 length-prefixed messages.
+- Commands (subset, as used):
+  - INIT, SHUT_DOWN, SUCCESS, NEXT_EVENT, ADVANCE_TIME, END
+  - ADD_NODE, UPDATE_NODE, REMOVE_NODE
+  - CONF_WIFI_RADIO, SEND_WIFI_MSG, RECV_WIFI_MSG
+  - CONF_CELL_RADIO, SEND_CELL_MSG, RECV_CELL_MSG
+
+| Command                         | Action                                                                                                           | Reply                                 |
+|---------------------------------|------------------------------------------------------------------------------------------------------------------|----------------------------------------|
+| INIT                            | Read InitMessage, check protocol and times.                                                                      | CMD_SUCCESS or CMD_SHUT_DOWN           |
+| ADD_NODE (RADIO/WIRED/NODE_B)   | Create radio node (UE + 802.11p), wired node (CSMA), or eNB; if after start, schedule activation.                | CMD_SUCCESS                            |
+| UPDATE_NODE                     | Update node positions (scheduled at given time).                                                                 | CMD_SUCCESS                            |
+| REMOVE_NODE                     | Disable node (turn off Wi-Fi PHY, disable apps).                                                                 | CMD_SUCCESS                            |
+| CONF_WIFI_RADIO                 | Enable Wi-Fi app, set TX power if provided, add Wi-Fi IP address to device.                                      | CMD_SUCCESS                            |
+| SEND_WIFI_MSG                   | Schedule UDP send via Wi-Fi app; channel/TTL ignored for now.                                                    | CMD_SUCCESS                            |
+| CONF_CELL_RADIO                 | Enable cell/CSMA app; add IP; for UEs attach to closest eNB; adjust routes for wired nodes.                      | CMD_SUCCESS                            |
+| SEND_CELL_MSG                   | Schedule UDP send via LTE (radio node) or CSMA (wired node).                                                     | CMD_SUCCESS                            |
+| SHUT_DOWN                       | Log stats, disable logging, destroy simulator, close loop.                                                       | —                                      |
+
+### Networking and routing notes
+- Backbone: CSMA at 100 Gb/s, PGW and Servers on this backbone
+- IP constraints:
+  - All node IPs must be within 10.0.0.0/8.
+  - Wired nodes must use 10.5.0.0/16 or 10.6.0.0/16; radio nodes must not.
+- Extra radio nodes are pre-created to allow activation later (limitation by ns-3).
+
+### Configuration and logging
```

Comment:
Add a point "- Each MOSAIC simulation scenario can bring their own `ns3_federate_config.xml` for scenario-specific configuration."